controllers/common/utils.go

```diff
@@ -218,6 +218,12 @@ func StringPtr(str string) *string {
 }
 
 func StringSliceEquals(x, y []string) bool {
+	if x == nil {
+		x = []string{}
+	}
+	if y == nil {
+		y = []string{}
+	}
```

Comment:
Adding nil checking for `x/y` removing the need to set IamInstanceProfile and SecurityGroupIds in tests.
It was failing as the arguments were `StringSliceEqual([]string{}, nil)` eg. https://github.com/eytan-avisror/instance-manager/pull/3/files#diff-6d52e99e4cfeaf5b9650f8e3992b44c55e4f2bff302917eb5fdbe2948c3104bcR218

Reply:
Good catch, i'll address it